onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1085,19 +1085,20 @@ void PoolOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Nod
   }
 
   const auto& output = node.OutputDefs()[0]->Name();
-  const auto& op = node.OpType();
+  const auto& op_type = node.OpType();
 
-  int32_t op_type;
-  if (op == "AveragePool" || op == "GlobalAveragePool")
-    op_type = ANEURALNETWORKS_AVERAGE_POOL_2D;
-  else  // (op == "MaxPool" || op == "GlobalMaxPool")
-    op_type = ANEURALNETWORKS_MAX_POOL_2D;
+  int32_t op_code;
+  bool is_average_pool = op_type == "AveragePool";
+  if (is_average_pool || op_type == "GlobalAveragePool")
+    op_code = ANEURALNETWORKS_AVERAGE_POOL_2D;
+  else  // (op_type == "MaxPool" || op_type == "GlobalMaxPool")
+    op_code = ANEURALNETWORKS_MAX_POOL_2D;
```

Comment:
As this occurs during model loading it would be nicer to explicitly check and have an 'else' branch for an unexpected op type

Reply:
Added check inside PoolOpBuilder::IsOpSupportedImpl, so else will not happen

---
In reply to: [459171046](https://github.com/microsoft/onnxruntime/pull/4582#discussion_r459171046) [](ancestors = 459171046)